cmd/oras/repository/tags.go

```diff
@@ -78,14 +78,36 @@ func showTags(opts showTagsOptions) error {
 	if err != nil {
 		return err
 	}
+	filter := ""
 	if opts.Reference != "" {
-		return fmt.Errorf("unexpected tag or digest %q found in repository reference %q", opts.Reference, opts.RawReference)
+		if isDigestTag(opts.Reference) {
```

Comment:
use [digest. Parse](https://pkg.go.dev/github.com/opencontainers/go-digest#Parse)(opts.Reference) to validate whether a reference was digest or not.  `isDigestTag` only checks if the tag is following [fallback schema for referrer API support](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#referrers-tag-schema) or not.